File: src/models/tokenizer/FuzzySearchTokenizer.java
package models.tokenizer;

import java.util.*;

public class FuzzySearchTokenizer implements Tokenizer {
    private static final int FUZZINESS = 1;
    private static List<Character> characters = new ArrayList<>();

    static {
        for (char c = 'a'; c <= 'z'; c++) {
            characters.add(c);
        }
        for (char c = '0'; c < '9'; c++) {
            characters.add(c);
        }
    }

    @Override
    public Set<String> tokenize(String text) {
        return new HashSet<>();
    }

    @Override
    public LinkedList<String> develop(String token) {
        LinkedList<String> developedTokens = new LinkedList<>(Collections.singletonList(token));
        for (int i = 0; i < FUZZINESS; i++) {
            LinkedList<String> tempDevelopedTokens = new LinkedList<>();
            for (String tok : developedTokens) {
                tempDevelopedTokens.addAll(insert(tok));
                tempDevelopedTokens.addAll(delete(tok));
                tempDevelopedTokens.addAll(substitute(tok));
            }
            developedTokens.addAll(tempDevelopedTokens);
            developedTokens.remove(token);
        }
        return developedTokens;
    }

    private LinkedList<String> delete(String token) {
        LinkedList<String> deletes = new LinkedList<>();

        for (int i = 0; i < token.length(); i++) {
            deletes.add(token.substring(0, i) + token.substring(i + 1));
        }
        return deletes;
    }

    private LinkedList<String> insert(String token) {
        LinkedList<String> inserts = new LinkedList<>();

        for (int i = 0; i <= token.length(); i++) {
            for (char c : characters) {
                if (i < token.length() && c == token.charAt(i)) continue;
                inserts.add(token.substring(0, i) + c + token.substring(i));
            }
        }
        return inserts;
    }

    private LinkedList<String> substitute(String token) {
        LinkedList<String> substitutes = new LinkedList<>();

        for (int i = 0; i < token.length(); i++) {
            for (char c : characters) {
                if (c == token.charAt(i)) continue;
                substitutes.add(token.substring(0, i) + c + token.substring(i + 1));
            }
        }
        return substitutes;
    }
}

File: C#/SimpleSearch/SimpleSearch/models/tokenizer/FuzzySearchTokenizer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SimpleSearch.models.tokenizer
{
    class FuzzySearchTokenizer : Tokenizer
    {
        private static readonly int FUZZINESS = 1;
        private static readonly IList<char> characters = new List<char>();

        static FuzzySearchTokenizer()
        {
            for (char c = 'a'; c <= 'z'; c++)
            {
                characters.Add(c);
            }
            for (char c = 'a'; c <= 'z'; c++)
            {
                characters.Add(c);
            }
        }

        public override ISet<string> Tokenize(string text) => new HashSet<string>();
        public override List<string> Develope(string token)
        {
            List<String> developedTokens = new List<string>(new string[] {token}.ToList());
            for (int i = 0; i < FUZZINESS; i++)
            {
                List<String> tempDevelopedTokens = new List<string>();
                foreach (string developedToken in developedTokens)
                {
                    tempDevelopedTokens.AddRange(insert(developedToken));
                    tempDevelopedTokens.AddRange(delete(developedToken));
                    tempDevelopedTokens.AddRange(substitute(developedToken));
                }
                developedTokens.AddRange(tempDevelopedTokens);
                developedTokens.Remove(token);
            }
            return developedTokens;
        }
        private LinkedList<String> delete(String token)
        {
            LinkedList<String> deletes = new LinkedList<string>();

            for (int i = 0; i < token.Length; i++)
            {
                deletes.AddLast(token.Substring(0, i) + token.Substring(i + 1));
            }
            return deletes;
        }

        private LinkedList<String> insert(String token)
        {
            LinkedList<String> inserts = new LinkedList<string>();

            for (int i = 0; i <= token.Length; i++)
            {
                foreach (char c in characters)
                {
                    if (i < token.Length && c == token[i]) continue;
                    inserts.AddLast(token.Substring(0, i) + c + token.Substring(i));
                }
            }
            return inserts;
        }

        private LinkedList<String> substitute(String token)
        {
            LinkedList<String> substitutes = new LinkedList<string>();

            for (int i = 0; i < token.Length; i++)
            {
                foreach (char c in characters)
                {
                    if (c == token[i]) continue;
                    substitutes.AddLast(token.Substring(0, i) + c + token.Substring(i + 1));
                }
            }
            return substitutes;
        }
    }
}
